net: mqtt: Allow to bind client to a specific interface

rlubos opened the pull request:
Add a new "if_name" pointer to the transport configuration structure, allowing the application to bind MQTT client to a specific network interface.

rlubos committed:
net: mqtt: Allow to bind client to a specific interface

Add a new "if_name" pointer to the transport configuration structure,
allowing the application to bind MQTT client to a specific network
interface.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

jukkar reviewed (dismissed):
LGTM. Is it possible to update/add tests for this?

rlubos committed:
tests: net: mqtt: client: Add test case for connect with iface binding

Add test case for verifying that MQTT client works fine if bound to an
interface.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

rlubos commented:
Added a simple test case.
